feat(tile-container): build tile container in web components storybook

devadula-nandan opened the pull request:
contributes #352 

build to publish `cds-aichat-tile-container` in ai-chat-components
add story showcasing the component

#### Changelog

**New**

- storybook 9 upgrades
- code tab, interactions tab
- new welcome page
- first component `tile-container`
- storybook 9 gives the ability to add transformer for the code tab, which we have used to strip style tags in the code tab.
- added `*.snap.js` to eslint ignore and prettier ignore, as we are noticing a weird problem where lint staged formatting the snap files, causing the snapshot matching to fail.
- added "aiChat:start:storybook" script command to root package.json to make it easier to start the storybook

**Changed**

- modified ci to build chat components before testing
- 

**Removed**

- removed old placeholder extended button

#### Testing / Reviewing

- make sure the components render as expected in the storybook
- review the docs page and the new welcome page
- make sure the interactions for clickable and ai-label variants work as expected

ethanwinters commented on packages/ai-chat-components/src/components/extended-button/src/extended-button.scss:
```diff
@@ -10,6 +10,6 @@ $css--plex: true !default;
 @use "../../../globals/scss/vars" as *;
 @use "@carbon/web-components/scss/components/button/button";
 
-:host(#{prefix}-extended-button) {
```
we can just delete the whole example "extended-button" now that we have a real component in here

ethanwinters commented on packages/ai-chat-components/src/react/tile-container.ts:
```diff
@@ -0,0 +1,22 @@
+  tagName: "cds-aichat-tile-container",
+  elementClass: CDSAIChatTileContainer,
+  react: React,
+});
```
So the built version of this points to "import { createComponent as o } from '../node_modules/@lit/react/create-component.js';"

I think we have to fix up the dependencies or ai-chat-components to include `@lit/react`

ethanwinters commented on packages/ai-chat/package.json:
```diff
@@ -46,6 +46,7 @@
     "**/web-components/**/*.js"
   ],
   "devDependencies": {
+    "@carbon/ai-chat-components": "*",
```
so something to keep in mind is that @carbon/ai-chat-components is currently "private". When do we make it public even if in beta?

devadula-nandan commented:
new updates:
upgraded storybook to 9
added a welcome page (need to review the content, maybe add hero image like what other repos did? designers? assets?)
added some interaction story tests as an exploration (we can disable them if we want)
there could be some issues with icon on image variant which would be discussed with the design later.
lets keep this as our first iteration of ai-chat tile component, and we can improve them with a11y and design inputs gradually.

ethanwinters commented:
Lets merge AFTER @annawen1 cuts 1.0.0.
